DQMOffline/Trigger/plugins/DiJetMonitor.cc

```diff
@@ -0,0 +1,294 @@
+  desc.add<edm::InputTag>( "electrons",edm::InputTag("gedGsfElectrons") );
+  desc.add<edm::InputTag>( "muons",    edm::InputTag("muons") );
+  desc.add<int>("njets",      0);
+  desc.add<int>("nelectrons", 0);
+  desc.add<double>("ptcut",   20);
+
+  edm::ParameterSetDescription genericTriggerEventPSet;
+  genericTriggerEventPSet.add<bool>("andOr");
+  genericTriggerEventPSet.add<edm::InputTag>("dcsInputTag", edm::InputTag("scalersRawToDigi") );
+  genericTriggerEventPSet.add<std::vector<int> >("dcsPartitions",{});
+  genericTriggerEventPSet.add<bool>("andOrDcs", false);
+  genericTriggerEventPSet.add<bool>("errorReplyDcs", true);
+  genericTriggerEventPSet.add<std::string>("dbLabel","");
+  genericTriggerEventPSet.add<bool>("andOrHlt", true);
+  genericTriggerEventPSet.add<edm::InputTag>("hltInputTag", edm::InputTag("TriggerResults::HLT") );
+  genericTriggerEventPSet.add<std::vector<std::string> >("hltPaths",{});
+  genericTriggerEventPSet.add<bool>("errorReplyHlt",false);
+  genericTriggerEventPSet.add<unsigned int>("verbosityLevel",1);
+
+  desc.add<edm::ParameterSetDescription>("numGenericTriggerEventPSet", genericTriggerEventPSet);
+  desc.add<edm::ParameterSetDescription>("denGenericTriggerEventPSet", genericTriggerEventPSet);
+
+  edm::ParameterSetDescription histoPSet;
+  edm::ParameterSetDescription dijetPSet;
+  edm::ParameterSetDescription dijetPtThrPSet;
+  fillHistoPSetDescription(dijetPSet);
+  fillHistoPSetDescription(dijetPtThrPSet);
+  histoPSet.add<edm::ParameterSetDescription>("dijetPSet", dijetPSet);  
+  histoPSet.add<edm::ParameterSetDescription>("dijetPtThrPSet", dijetPtThrPSet);  
+  std::vector<double> bins = {0.,20.,40.,60.,80.,90.,100.,110.,120.,130.,140.,150.,160.,170.,180.,190.,200.,220.,240.,260.,280.,300.,350.,400.,450.,1000.}; // DiJet pT Binning
+  histoPSet.add<std::vector<double> >("jetptBinning", bins);
+
+  edm::ParameterSetDescription lsPSet;
+  fillHistoLSPSetDescription(lsPSet);
+  histoPSet.add<edm::ParameterSetDescription>("lsPSet", lsPSet);
+  desc.add<edm::ParameterSetDescription>("histoPSet",histoPSet);
+
+  descriptions.add("dijetMonitoring", desc);
+}
+
+//---- Additional DiJet offline selection------
+bool DiJetMonitor::dijet_selection(double eta_1, double phi_1, double eta_2, double phi_2, double pt_1, double pt_2, int &tag_id, int &probe_id){
+  double etacut_ = 1.7;
+  double phicut_ = 2.7;
+	
+  bool passeta = false; //check that one of the jets in the barrel
+  if (abs(eta_1)< etacut_ || abs(eta_2) < etacut_ )  passeta=true;
+  
+  float delta_phi_1_2= (phi_1 - phi_2);
+  bool other_cuts = false;//check that jets are back to back
+  if (abs(delta_phi_1_2) >= phicut_) other_cuts=true;
+
+   if(fabs(eta_1)<etacut_ && fabs(eta_2)>etacut_) {
+    tag_id = 0; 
+    probe_id = 1;
+  }
+  else 
+    if(fabs(eta_2)<etacut_ && fabs(eta_1)>etacut_) {
+      tag_id = 1; 
+      probe_id = 0;
+    }
+    else 
+      if(fabs(eta_2)<etacut_ && fabs(eta_1)<etacut_){
+    int ran = rand();
```

Comment:
you need a reproducible method here. (so avoid random numbers). I'm not really sure why you can't independently use each leg as a tag?

Reply:
Dear David,
I'm sorry for late replying.
I received this offline selection from Anastasia for JEC.
If you want me to change this code, I'll ask Anastasia.
Thank you. :)